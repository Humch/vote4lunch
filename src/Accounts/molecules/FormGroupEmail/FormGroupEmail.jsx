import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import {
  InputGroup,
  InputGroupAddon,
  InputGroupText,
} from 'reactstrap';

import { Label, FormFeedback } from '../../../Core';
import { FormInputEmail, EmailChecker } from '../..';

import { checkEmail } from '../../actions';

class FormGroupEmail extends Component {
  componentDidUpdate(prevProps) {
    const { email, checkEmail: checkE } = this.props;
    if (prevProps.email !== email) {
      checkE(email);
    }
  }

  render() {
    const {
      email,
      text,
      forProp,
      colorLabel
    } = this.props;

    let rendering = '';

    if (email !== '') {
      rendering = (
        <InputGroupAddon addonType="append">
          <InputGroupText>
            <EmailChecker />
          </InputGroupText>
        </InputGroupAddon>
      );
    }

    return (
      <div className="form-group">
        <Label
          text={text}
          forProp={forProp}
          color={colorLabel}
        />
        <InputGroup>
          <FormInputEmail />
          { rendering }
          <FormFeedback
            text="Cet email est déjà utilisé"
          />
        </InputGroup>
      </div>
    );
  }
}

FormGroupEmail.propTypes = {
  checkEmail: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  forProp: PropTypes.string.isRequired,
  colorLabel: PropTypes.string,
};

const mstp = ({ userData }) => ({
  email: userData.email,
});

const mdtp = dispatch => bindActionCreators({ checkEmail }, dispatch);

export default connect(mstp, mdtp)(FormGroupEmail);
