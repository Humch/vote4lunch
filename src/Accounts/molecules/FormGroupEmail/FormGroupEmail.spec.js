/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';

import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store';

import FormGroupEmail from './FormGroupEmail';

describe('FormGroupEmail Snapshot', () => {
  const initialState = {
    userData: {
      email: '',
    },
    emailChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const text = 'le super label';
  const forProp = 'email';
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormGroupEmail', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <FormGroupEmail
          text={text}
          forProp={forProp}
        />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('FormGoupEmail Snapshot', () => {
  const initialState = {
    userData: {
      email: 'bob@bob.com',
    },
    emailChecker: {
      payload: { exist: false },
      loading: false,
    }
  };
  const text = 'le super label';
  const forProp = 'email';
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of FormGroupEmail with email', () => {
    const renderedValue = renderer.create(
      <Provider store={store}>
        <FormGroupEmail
          text={text}
          forProp={forProp}
        />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('FormGroupEmail - REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = {
    userData: {
      email: '',
    },
    emailChecker: {
      payload: {
        exist: false,
      },
    },
  };
  const text = 'le super label';
  const forProp = 'email';
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('render the connected(SMART) component', () => {
    wrapper = mount(
      <Provider store={store}>
        <FormGroupEmail
          text={text}
          forProp={forProp}
        />
      </Provider>
    );
    expect(wrapper.length).toEqual(1);
  });

  it('expect the input have is-invalid className', () => {
    initialState.emailChecker.payload.exist = true;
    wrapper = mount(
      <Provider store={store}>
        <FormGroupEmail
          text={text}
          forProp={forProp}
        />
      </Provider>
    );
    const input = wrapper.find('input');

    expect(input.hasClass('is-invalid')).toBe(true);
  });
});
