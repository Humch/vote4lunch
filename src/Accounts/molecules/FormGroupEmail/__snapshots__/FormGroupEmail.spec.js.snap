// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGoupEmail Snapshot capturing Snapshot of FormGroupEmail with email 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="email"
  >
    le super label
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control "
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value="bob@bob.com"
    />
    <div
      className="input-group-append"
    >
      <span
        className="input-group-text"
      />
    </div>
    <div
      className="invalid-feedback"
    >
      Cet email est déjà utilisé
    </div>
  </div>
</div>
`;

exports[`FormGroupEmail Snapshot capturing Snapshot of FormGroupEmail 1`] = `
<div
  className="form-group"
>
  <label
    className=""
    htmlFor="email"
  >
    le super label
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control "
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    
    <div
      className="invalid-feedback"
    >
      Cet email est déjà utilisé
    </div>
  </div>
</div>
`;
